core/drivers/scmi-msg/entry.c

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2015-2019, Arm Limited and Contributors. All rights reserved.
+ * Copyright (c) 2019, Linaro Limited
+ */
+#include <speculation_barrier.h>
+#include <drivers/scmi-msg.h>
+#include <drivers/scmi.h>
+#include <trace.h>
+
+#include "base.h"
+#include "common.h"
+
+scmi_msg_handler_t scmi_get_msg_handler(unsigned int message_id,
+					const scmi_msg_handler_t *handler_table,
+					size_t elt_count)
+{
+	/* Cast discards const qualifier to ease code readibility */
```

Comment:
``` C
        const scmi_msg_handler_t *min = handler_table;
        const scmi_msg_handler_t *max = min + elt_count;
        const scmi_msg_handler_t *ptr = min + message_id;
```
doesn't seem much harder to read...

Reply:
:)

Reply:
Tried, but fails to build due to load_no_speculate_fail() implementation:
```shell
core/drivers/scmi-msg/entry.c: In function ‘scmi_get_msg_handler’:
lib/libutils/ext/include/speculation_barrier.h:189:7: error: read-only variable ‘__nln_val’ used as ‘asm’ output
       __asm__ volatile (      \
       ^~~~~~~
```